CIPs/cip-0021.md

```diff
@@ -35,36 +35,37 @@ In practice, the lookback window has penalized validators for performing routine
 ## Goals
 
 - The lookback window parameter must be adjustable.
+- The lookback window parameter must be constant during an epoch
 - The community must decide on the value.
 
 ## Specification
 
 The proposed solution is to migrate the parameter to the EVM state (layer 2).
 
-We define the following constants:
+Starting at the first epoch after `HARDFORK_BLOCK`, block creation and validation MUST read the lookback window from chain state by calling the `BlockchainParameters` contract, rather than reading chain paramaters. Reward calculations are unaffected.
```

Comment:
```suggestion
Starting at the first full post-fork epoch (i.e. the same epoch as `HARDFORK_BLOCK` if it's the first block of its epoch, or the next epoch if it isn't), block creation and validation MUST read the lookback window from chain state by calling the `BlockchainParameters` contract, rather than using the value from the genesis config. Reward calculations are unaffected.
```

This change makes it correct also for the edge case where `HARDFORK_BLOCK` happens to be the first block on an epoch.